controllers/helmchartinstall_controller.go

```diff
@@ -77,61 +126,255 @@ func (r *HelmChartInstallReconciler) render(ctx context.Context, claim *helmv1al
+	configuration *v1alpha1.StackConfiguration,
+) (*v1alpha1.StackConfigurationBehavior, error) {
+	targetResourceGroupVersion, targetResourceKind := claim.GetObjectKind().GroupVersionKind().ToAPIVersionAndKind()
+	targetGroupKindVersion := fmt.Sprintf("%s.%s", targetResourceKind, targetResourceGroupVersion)
+
+	var targetResourceBehavior v1alpha1.StackConfigurationBehavior
+	// TODO handle missing keys gracefully
+	targetResourceBehavior, ok := configuration.Spec.Behaviors.CRDs[targetGroupKindVersion]
+
+	if !ok {
+		// TODO error condition with a real error returned
+		r.Log.V(0).Info("Couldn't find a configured behavior!",
+			"claim", claim,
+			"configuration", configuration,
+			"targetGroupKindVersion", targetGroupKindVersion,
+		)
+		return nil, nil
+
+	}
+
+	if len(targetResourceBehavior.Resources) == 0 {
+		// TODO error condition with a real error returned
+		// TODO it'd be nice to enforce this on acceptance or creation if possible
+		r.Log.V(0).Info("Couldn't find resources for configured behavior!", "claim", claim, "configuration", configuration)
+		return nil, nil
+
+	}
+
+	return &targetResourceBehavior, nil
+}
+
+/**
+ * When a behavior executes, the resource engine is configured by the
+ * object which triggered the behavior. This method encapsulates the logic to
+ * create the resource engine configuration from the object's fields.
+ *
+ * TODO Currently creation fails if the config map already exists, but it should
+ * succeed instead.
+ */
+func (r *HelmChartInstallReconciler) createBehaviorEngineConfiguration(
+	ctx context.Context,
+	claim *unstructured.Unstructured,
+	stackConfig *v1alpha1.StackConfiguration,
+) (*corev1.ConfigMap, error) {
+	// yamlyamlyamlyamlyaml
+	// TODO if spec is missing, that won't work very well
+	spec, ok := claim.Object["spec"]
+
+	if !ok {
+		r.Log.V(0).Info("Spec not found on claim; not creating engine configuration", "claim", claim)
+	}
+
+	r.Log.V(0).Info("Converting configuration", "spec", spec)
+	configContents, err := yaml.Marshal(spec)
+
+	r.Log.V(0).Info("Configuration contents as yaml", "configContents", configContents)
+
+	if err != nil {
+		r.Log.Error(err, "Error marshaling claim spec as yaml!", "claim", claim)
+		return nil, err
+	}
+
+	// Underneath, the yamler uses https://godoc.org/encoding/json#Marshal,
+	// which means that the bytes are UTF-8 encoded
+	// Theoretically we could get better performance by using a binary config
+	// map, but having a string makes it better for humans who may want to observe
+	// or troubleshoot behavior.
+	stringConfigContents := string(configContents)
+
+	// TODO get the engine type from the configuration
+	engineType := r.getEngineType(claim, stackConfig)
+
+	// TODO engine type should have a bit more structure;
+	// probably better to use an enum type pattern, with an
+	// engine name and its corresponding configuration file
+	// name in the same object
+	configKeyName := ""
+
+	if engineType == "helm2" {
+		configKeyName = "values.yaml"
+	}
```

Comment:
Should these be constants?

Reply:
I decided to leave this as-is for now, since I'm not confident this will be the structure we want in the long run, and if the code is bad it'll motivate me to fix it in the future.